### PR TITLE
feat(github-actions): support workflow templates

### PR DESCRIPTION
**Pull request by bendrucker**
## Changes:

This PR adds support for updating [GitHub Actions workflow templates](https://docs.github.com/en/actions/learn-github-actions/sharing-workflows-with-your-organization). Workflow templates are essentially a normal actions workflow file, but with a `$default-branch` template variable that will be substituted when the template is used in a repo. 

## Documentation

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/manager/github-actions/index.ts [hunk 1]
```diff
@@ -6,5 +6,5 @@ const language = LANGUAGE_DOCKER;
 export { extractPackageFile, language };
 
 export const defaultConfig = {
-  fileMatch: ['^\\.github\\/workflows\\/[^/]+\\.ya?ml$'],
+  fileMatch: ['^\\.github\\/workflows|workflow-templates\\/[^/]+\\.ya?ml$'],
```
Comment (rarkins) [inline lib/manager/github-actions/index.ts:9]:
```suggestion
  fileMatch: ['^\\.github\\/(workflows|workflow-templates)\\/[^/]+\\.ya?ml$'],
```

**Review by rarkins — CHANGES_REQUESTED** [review]
This should be tested on a repo first

**Comment by bendrucker** [comment]
Definitely did need that repo test! 😬 Forgot these are stored as `workflow-templates` in a `.github` repository, but not in a `.github` directory. Fixed the regex and tested:

```
yarn start bendrucker/renovate-test
yarn run v1.10.1
$ run-s generate:*
$ node tools/generate-imports.mjs
generating imports
> data/extract.py
> data/node-js-schedule.json
$ node -r ts-node/register/transpile-only -- lib/renovate.ts bendrucker/renovate-test
 INFO: Repository started (repository=bendrucker/renovate-test)
       "renovateVersion": "0.0.0-semantic-release"
 INFO: Branch created (repository=bendrucker/renovate-test, branch=renovate/configure)
       "commit": "b49d70a",
       "onboarding": true
 INFO: Dependency extraction complete (repository=bendrucker/renovate-test)
       "baseBranch": "main",
       "stats": {
         "managers": {"github-actions": {"fileCount": 1, "depCount": 2}},
         "total": {"fileCount": 1, "depCount": 2}
       }
 INFO: Onboarding PR created (repository=bendrucker/renovate-test)
       "pr": "Pull Request #1"
 INFO: Repository finished (repository=bendrucker/renovate-test)
       "durationMs": 12482
✨  Done in 22.92s.
```

With that fix in place, the package file is detected.

**Comment by bendrucker** [comment]
Apologies for the extra cycle, thanks for providing such thorough developer docs!

**Review by monstro805 — APPROVED** [review]

**Review by viceice — APPROVED** [review]

File: lib/manager/github-actions/index.ts (hunk 1 above)
Comment (viceice) [reply lib/manager/github-actions/index.ts]: The folder needs to be in repo root of `.github` repo according to the docs

**Review by rarkins — APPROVED** [review]

**Pull request merged**